zookeeper-server/src/main/java/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -151,12 +148,17 @@ void sendBufferSync(ByteBuffer bb) {
      * sendBuffer pushes a byte buffer onto the outgoing buffer queue for
      * asynchronous writes.
      */
-    public void sendBuffer(ByteBuffer bb) {
+    public void sendBuffer(ByteBuffer... buffers) {
         if (LOG.isTraceEnabled()) {
             LOG.trace("Add a buffer to outgoingBuffers, sk " + sk
                       + " is valid: " + sk.isValid());
         }
-        outgoingBuffers.add(bb);
+        synchronized (outgoingBuffers) {
```

Comment:
This synchronization had to be added, because of the ability to send multiple byte buffers at once. What's the performance impact?